graphql_compiler/query_pagination/__init__.py

```diff
@@ -1,98 +1,114 @@
     #             print the given AST and provide that to the cost estimator.
-    graphql_query_string = print_ast(query_ast)
-    num_pages = estimate_number_of_pages(schema_info, graphql_query_string, parameters, page_size)
+    graphql_query_string = print_ast(query.query_ast)
+    num_pages = estimate_number_of_pages(
+        schema_info, graphql_query_string, query.parameters, page_size
+    )
     if num_pages > 1:
-        result_queries = split_into_page_query_and_remainder_query(
-            schema_info, ASTWithParameters(query_ast, parameters), num_pages
+        page_query, remainder_query, advisories = split_into_page_query_and_remainder_query(
+            schema_info, query, num_pages
         )
+        remainder_queries = (remainder_query,)
 
-    return result_queries
+    return page_query, remainder_queries, advisories
 
 
-def paginate_query(schema_info, query_string, parameters, page_size):
-    """Generate a query fetching a page of results and the remainder query for a query string.
+def paginate_query(
+    schema_info: QueryPlanningSchemaInfo, query: QueryStringWithParameters, page_size: int
+) -> Tuple[
+    QueryStringWithParameters, Tuple[QueryStringWithParameters, ...], Tuple[PaginationAdvisory, ...]
+]:
+    """Generate a query fetching a page of results and the remainder queries for a query string.
 
     Since the cost estimator may underestimate or overestimate the actual number of pages, you
     should expect the actual number of results of the page query to be within two orders of
     magnitude of the estimate.
 
     Args:
         schema_info: QueryPlanningSchemaInfo
-        schema_graph: SchemaGraph instance.
-        statistics: Statistics object.
-        query_string: str, valid GraphQL query to be paginated.
+        query: QueryStringWithParameters
         parameters: dict, parameters with which query will be estimated.
         page_size: int, describes the desired number of result rows per page.
 
     Returns:
         tuple containing queries for going over the original query in a paginated fashion:
-            - QueryStringWithParameters namedtuple, query expected to return roughly a page of
+            - QueryStringWithParameters, query expected to return roughly a page of
               result data of the original query.
-            - QueryStringWithParameters namedtuple or None. If the given query was estimated to
-              return more than a page of results, this element is a QueryStringWithParameters
-              namedtuple describing a query for the rest of the result data. Otherwise, this element
-              is None.
+            - Tuple of QueryStringWithParameters, describing queries that return the rest of the
+              result data of the original query. If the original query is expected to return only
+              a page or less of results, then this element will be an empty tuple. If the pagination
```

Comment:
Bojan thanks for the explanation. I now understand the purpose of this diff.

I would remove the reference to the internal pagination internals: `VertexPartitionPlans`. If you think that it is too hard to explain the reasoning without this reference, then feel free to delete this explanation.